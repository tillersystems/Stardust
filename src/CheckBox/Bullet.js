import React from 'react';
import PropTypes from 'prop-types';

import Icon from '../Icon';

import { StyledCheckbox } from './elements';

const CheckboxBullet = ({ isChecked }) => (
  <StyledCheckbox className={isChecked ? 'checked' : ''} data-testid="styled-checkBox">
    <Icon name="check-mark" color="currentColor" width="10px" height="10px" />
  </StyledCheckbox>
);

CheckboxBullet.propTypes = {
  isChecked: PropTypes.bool.isRequired,
};

export default CheckboxBullet;
