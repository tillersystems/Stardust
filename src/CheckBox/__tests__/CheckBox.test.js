import React from 'react';
import { fireEvent } from 'react-testing-library';

import CheckBox from '..';

describe('<CheckBox />', () => {
  test('should render without a problem', () => {
    const { container } = render(<CheckBox>label</CheckBox>);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render an initial selected checkbox', () => {
    const { container } = render(<CheckBox checked>label</CheckBox>);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('should be disabled', () => {
    const { getByTestId } = render(<CheckBox disabled>label</CheckBox>);
    const checkboxNode = getByTestId('checkBoxContainer');

    expect(checkboxNode).toHaveStyleRule('opacity', '0.4');
    expect(checkboxNode).toHaveStyleRule('cursor', 'not-allowed');
  });

  test('should be checked on change', () => {
    const { getByLabelText } = render(<CheckBox>label</CheckBox>);
    const checkboxNode = getByLabelText('label');

    fireEvent.click(checkboxNode);

    expect(checkboxNode).toMatchSnapshot();
  });

  test('should not be checked with change when disabled', () => {
    const spy = jest.fn();
    const { container, getByLabelText } = render(
      <CheckBox disabled onChange={spy}>
        label
      </CheckBox>,
    );
    const checkboxLabelNode = getByLabelText('label');

    fireEvent.click(checkboxLabelNode);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('should call onChange callback', () => {
    const spy = jest.fn();
    const { getByLabelText } = render(<CheckBox onChange={spy}>label</CheckBox>);
    const checkboxLabelNode = getByLabelText('label');

    fireEvent.click(checkboxLabelNode);

    expect(spy).toHaveBeenCalledTimes(1);
  });

  // test('should stopPropagation on input to prevents clicking on the label trigging the event twice', () => {
  //   const spy = jest.fn();
  //   const { getByTestId } = render(<CheckBox onChange={spy}>label</CheckBox>);
  //   const checkboxInputNode = getByTestId('checkBoxInput');
  //
  //   fireEvent.click(checkboxInputNode);
  //
  //   expect(spy).toHaveBeenCalledTimes(0);
  // });
});
