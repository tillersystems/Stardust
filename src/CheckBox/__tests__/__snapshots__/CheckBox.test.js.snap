// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBox /> should render without a problem 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 1.4rem;
  height: 1.4rem;
  border: 1px solid hsl(200,74%,46%);
  border-radius: 0.2rem;
  background: hsl(0,100%,100%);
}

.c2 {
  display: block;
  margin-left: 0.7rem;
  color: hsl(213,17%,20%);
}

<ThemeProvider
  theme={
    Object {
      "breakpoints": Object {
        "lg": 992,
        "md": 768,
        "sm": 576,
        "xl": 1200,
        "xs": 0,
      },
      "dimensions": Object {
        "big": "3.1rem",
        "bigInt": 3.1,
        "huge": "4.5rem",
        "hugeInt": 4.5,
        "medium": "1.5rem",
        "mediumInt": 1.5,
        "radius": "0.4rem",
        "radiusInt": 0.4,
        "small": "0.7rem",
        "smallInt": 0.7,
        "tiny": "0.3rem",
        "tinyInt": 0.3,
      },
      "fonts": Object {
        "family": "\\"Open Sans\\", Helvetica, Arial, sans-serif",
        "size": Object {
          "big": "3.2rem",
          "bigInt": 3.2,
          "huge": "6.4rem",
          "hugeInt": 6.4,
          "medium": "1.2rem",
          "mediumInt": 1.2,
          "small": "0.7rem",
          "smallInt": 0.7,
          "tiny": "0.3rem",
          "tinyInt": 0.3,
        },
        "spacing": Object {
          "big": "1.5px",
          "medium": "1px",
          "small": "0.5px",
        },
        "weight": Object {
          "normal": 400,
          "thick": 600,
          "thin": 200,
        },
      },
      "palette": Object {
        "alizarin": "hsl(6, 78%, 57%)",
        "anthracite": "hsl(213, 17%, 20%)",
        "black": "hsl(0, 0%, 0%)",
        "blackLight": "hsl(213, 17%, 20%)",
        "blue": "hsl(200, 74%, 46%)",
        "bodyBackground": "hsl(218, 29%, 95%)",
        "carrot": "hsl(28, 80%, 52%)",
        "clay": "hsl(215, 17%, 26%)",
        "darkBlue": "#1c88bf",
        "darkGray": "#dbdfe6",
        "darkGreen": "#7fbf3b",
        "darkRed": "#ea6151",
        "emerald": "hsl(145, 63%, 49%)",
        "fadeBlue": "hsl(210, 50%, 98%)",
        "failure": Object {
          "back": "hsl(6, 79%, 65%)",
          "darkBack": "#ea6151",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#f1978d",
          "lightFore": "#fff",
        },
        "googleBrandRed": "#ea4335",
        "gray": "hsl(218, 18%, 91%)",
        "green": "hsl(89, 53%, 52%)",
        "grey": "hsl(206, 16%, 60%)",
        "greygoose": "hsl(216, 29%, 97%)",
        "lightBlue": "#3baae2",
        "lightGray": "#fff",
        "lightGreen": "#a0d16b",
        "lightRed": "#f1978d",
        "marble": "hsl(206, 23%, 69%)",
        "orange": "hsl(34, 91%, 65%)",
        "primary": Object {
          "back": "hsl(200, 74%, 46%)",
          "darkBack": "#1c88bf",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#3baae2",
          "lightFore": "#fff",
        },
        "red": "hsl(6, 79%, 65%)",
        "river": "hsl(204, 70%, 53%)",
        "sand": "hsl(214, 8%, 84%)",
        "secondary": Object {
          "back": "hsl(0, 100%, 100%)",
          "darkBack": "#f7f7f7",
          "darkFore": "#8093a3",
          "fore": "hsl(206, 16%, 60%)",
          "lightBack": "#fff",
          "lightFore": "#a6b4be",
        },
        "success": Object {
          "back": "hsl(89, 53%, 52%)",
          "darkBack": "#7fbf3b",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#a0d16b",
          "lightFore": "#fff",
        },
        "turquoise": "hsl(168, 76%, 42%)",
        "white": "hsl(0, 100%, 100%)",
        "wisteria": "hsl(282, 44%, 47%)",
      },
    }
  }
>
  <CheckBox
    checked={false}
    disabled={false}
    id="test"
    onChange={null}
  >
    <styled.div
      onClick={[Function]}
    >
      <div
        className="c0"
        onClick={[Function]}
      >
        <styled.div
          checked={false}
          disabled={false}
          hasFocus={false}
        >
          <div
            checked={false}
            className="c1"
            disabled={false}
          >
            <input
              checked={false}
              disabled={false}
              id="test"
              onBlur={[Function]}
              onFocus={[Function]}
              tabIndex="0"
              type="checkbox"
            />
          </div>
        </styled.div>
        <styled.label
          disabled={false}
          hasFocus={false}
          htmlFor="test"
        >
          <label
            className="c2"
            disabled={false}
            htmlFor="test"
          />
        </styled.label>
      </div>
    </styled.div>
  </CheckBox>
</ThemeProvider>
`;

exports[`<CheckBox /> should render without a problem when checked and disabled 1`] = `
.c2 {
  display: inline-block;
  font-size: 1rem;
  line-height: 1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 1.4rem;
  height: 1.4rem;
  border: 1px solid hsl(200,74%,46%);
  border-radius: 0.2rem;
  background: hsl(0,100%,100%);
  opacity: 0.4;
}

.c3 {
  display: block;
  margin-left: 0.7rem;
  color: hsl(213,17%,20%);
  opacity: 0.4;
}

<ThemeProvider
  theme={
    Object {
      "breakpoints": Object {
        "lg": 992,
        "md": 768,
        "sm": 576,
        "xl": 1200,
        "xs": 0,
      },
      "dimensions": Object {
        "big": "3.1rem",
        "bigInt": 3.1,
        "huge": "4.5rem",
        "hugeInt": 4.5,
        "medium": "1.5rem",
        "mediumInt": 1.5,
        "radius": "0.4rem",
        "radiusInt": 0.4,
        "small": "0.7rem",
        "smallInt": 0.7,
        "tiny": "0.3rem",
        "tinyInt": 0.3,
      },
      "fonts": Object {
        "family": "\\"Open Sans\\", Helvetica, Arial, sans-serif",
        "size": Object {
          "big": "3.2rem",
          "bigInt": 3.2,
          "huge": "6.4rem",
          "hugeInt": 6.4,
          "medium": "1.2rem",
          "mediumInt": 1.2,
          "small": "0.7rem",
          "smallInt": 0.7,
          "tiny": "0.3rem",
          "tinyInt": 0.3,
        },
        "spacing": Object {
          "big": "1.5px",
          "medium": "1px",
          "small": "0.5px",
        },
        "weight": Object {
          "normal": 400,
          "thick": 600,
          "thin": 200,
        },
      },
      "palette": Object {
        "alizarin": "hsl(6, 78%, 57%)",
        "anthracite": "hsl(213, 17%, 20%)",
        "black": "hsl(0, 0%, 0%)",
        "blackLight": "hsl(213, 17%, 20%)",
        "blue": "hsl(200, 74%, 46%)",
        "bodyBackground": "hsl(218, 29%, 95%)",
        "carrot": "hsl(28, 80%, 52%)",
        "clay": "hsl(215, 17%, 26%)",
        "darkBlue": "#1c88bf",
        "darkGray": "#dbdfe6",
        "darkGreen": "#7fbf3b",
        "darkRed": "#ea6151",
        "emerald": "hsl(145, 63%, 49%)",
        "fadeBlue": "hsl(210, 50%, 98%)",
        "failure": Object {
          "back": "hsl(6, 79%, 65%)",
          "darkBack": "#ea6151",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#f1978d",
          "lightFore": "#fff",
        },
        "googleBrandRed": "#ea4335",
        "gray": "hsl(218, 18%, 91%)",
        "green": "hsl(89, 53%, 52%)",
        "grey": "hsl(206, 16%, 60%)",
        "greygoose": "hsl(216, 29%, 97%)",
        "lightBlue": "#3baae2",
        "lightGray": "#fff",
        "lightGreen": "#a0d16b",
        "lightRed": "#f1978d",
        "marble": "hsl(206, 23%, 69%)",
        "orange": "hsl(34, 91%, 65%)",
        "primary": Object {
          "back": "hsl(200, 74%, 46%)",
          "darkBack": "#1c88bf",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#3baae2",
          "lightFore": "#fff",
        },
        "red": "hsl(6, 79%, 65%)",
        "river": "hsl(204, 70%, 53%)",
        "sand": "hsl(214, 8%, 84%)",
        "secondary": Object {
          "back": "hsl(0, 100%, 100%)",
          "darkBack": "#f7f7f7",
          "darkFore": "#8093a3",
          "fore": "hsl(206, 16%, 60%)",
          "lightBack": "#fff",
          "lightFore": "#a6b4be",
        },
        "success": Object {
          "back": "hsl(89, 53%, 52%)",
          "darkBack": "#7fbf3b",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#a0d16b",
          "lightFore": "#fff",
        },
        "turquoise": "hsl(168, 76%, 42%)",
        "white": "hsl(0, 100%, 100%)",
        "wisteria": "hsl(282, 44%, 47%)",
      },
    }
  }
>
  <CheckBox
    checked={true}
    disabled={true}
    id="test"
    onChange={null}
  >
    <styled.div
      onClick={[Function]}
    >
      <div
        className="c0"
        onClick={[Function]}
      >
        <styled.div
          checked={true}
          disabled={true}
          hasFocus={false}
        >
          <div
            checked={true}
            className="c1"
            disabled={true}
          >
            <Styled(Icon)
              color="hsl(200, 74%, 46%)"
              height="1rem"
              name="check-mark"
              width="1rem"
            >
              <Icon
                className=""
                color="hsl(200, 74%, 46%)"
                height="1rem"
                name="check-mark"
                spin={false}
                width="1rem"
              >
                <styled.div
                  spin={false}
                >
                  <div
                    className="c2"
                  >
                    <svg
                      height="1rem"
                      viewBox="0 0 512 512"
                      width="1rem"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M411.582 134.248c8.33-8.33 21.838-8.33 30.17 0 8.33 8.332 8.33 21.84 0 30.17L207.085 399.085c-8.331 8.331-21.839 8.331-30.17 0L70.248 292.418c-8.33-8.33-8.33-21.838 0-30.17 8.332-8.33 21.84-8.33 30.17 0L192 353.83l219.582-219.582z"
                        fill="hsl(200, 74%, 46%)"
                        fillRule="evenodd"
                      />
                    </svg>
                  </div>
                </styled.div>
              </Icon>
            </Styled(Icon)>
            <input
              checked={true}
              disabled={true}
              id="test"
              onBlur={[Function]}
              onFocus={[Function]}
              tabIndex="0"
              type="checkbox"
            />
          </div>
        </styled.div>
        <styled.label
          disabled={true}
          hasFocus={false}
          htmlFor="test"
        >
          <label
            className="c3"
            disabled={true}
            htmlFor="test"
          />
        </styled.label>
      </div>
    </styled.div>
  </CheckBox>
</ThemeProvider>
`;

exports[`<CheckBox /> should render without a problem when checked and enabled 1`] = `
.c2 {
  display: inline-block;
  font-size: 1rem;
  line-height: 1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 1.4rem;
  height: 1.4rem;
  border: 1px solid hsl(200,74%,46%);
  border-radius: 0.2rem;
  background: hsl(0,100%,100%);
}

.c3 {
  display: block;
  margin-left: 0.7rem;
  color: hsl(213,17%,20%);
}

<ThemeProvider
  theme={
    Object {
      "breakpoints": Object {
        "lg": 992,
        "md": 768,
        "sm": 576,
        "xl": 1200,
        "xs": 0,
      },
      "dimensions": Object {
        "big": "3.1rem",
        "bigInt": 3.1,
        "huge": "4.5rem",
        "hugeInt": 4.5,
        "medium": "1.5rem",
        "mediumInt": 1.5,
        "radius": "0.4rem",
        "radiusInt": 0.4,
        "small": "0.7rem",
        "smallInt": 0.7,
        "tiny": "0.3rem",
        "tinyInt": 0.3,
      },
      "fonts": Object {
        "family": "\\"Open Sans\\", Helvetica, Arial, sans-serif",
        "size": Object {
          "big": "3.2rem",
          "bigInt": 3.2,
          "huge": "6.4rem",
          "hugeInt": 6.4,
          "medium": "1.2rem",
          "mediumInt": 1.2,
          "small": "0.7rem",
          "smallInt": 0.7,
          "tiny": "0.3rem",
          "tinyInt": 0.3,
        },
        "spacing": Object {
          "big": "1.5px",
          "medium": "1px",
          "small": "0.5px",
        },
        "weight": Object {
          "normal": 400,
          "thick": 600,
          "thin": 200,
        },
      },
      "palette": Object {
        "alizarin": "hsl(6, 78%, 57%)",
        "anthracite": "hsl(213, 17%, 20%)",
        "black": "hsl(0, 0%, 0%)",
        "blackLight": "hsl(213, 17%, 20%)",
        "blue": "hsl(200, 74%, 46%)",
        "bodyBackground": "hsl(218, 29%, 95%)",
        "carrot": "hsl(28, 80%, 52%)",
        "clay": "hsl(215, 17%, 26%)",
        "darkBlue": "#1c88bf",
        "darkGray": "#dbdfe6",
        "darkGreen": "#7fbf3b",
        "darkRed": "#ea6151",
        "emerald": "hsl(145, 63%, 49%)",
        "fadeBlue": "hsl(210, 50%, 98%)",
        "failure": Object {
          "back": "hsl(6, 79%, 65%)",
          "darkBack": "#ea6151",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#f1978d",
          "lightFore": "#fff",
        },
        "googleBrandRed": "#ea4335",
        "gray": "hsl(218, 18%, 91%)",
        "green": "hsl(89, 53%, 52%)",
        "grey": "hsl(206, 16%, 60%)",
        "greygoose": "hsl(216, 29%, 97%)",
        "lightBlue": "#3baae2",
        "lightGray": "#fff",
        "lightGreen": "#a0d16b",
        "lightRed": "#f1978d",
        "marble": "hsl(206, 23%, 69%)",
        "orange": "hsl(34, 91%, 65%)",
        "primary": Object {
          "back": "hsl(200, 74%, 46%)",
          "darkBack": "#1c88bf",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#3baae2",
          "lightFore": "#fff",
        },
        "red": "hsl(6, 79%, 65%)",
        "river": "hsl(204, 70%, 53%)",
        "sand": "hsl(214, 8%, 84%)",
        "secondary": Object {
          "back": "hsl(0, 100%, 100%)",
          "darkBack": "#f7f7f7",
          "darkFore": "#8093a3",
          "fore": "hsl(206, 16%, 60%)",
          "lightBack": "#fff",
          "lightFore": "#a6b4be",
        },
        "success": Object {
          "back": "hsl(89, 53%, 52%)",
          "darkBack": "#7fbf3b",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#a0d16b",
          "lightFore": "#fff",
        },
        "turquoise": "hsl(168, 76%, 42%)",
        "white": "hsl(0, 100%, 100%)",
        "wisteria": "hsl(282, 44%, 47%)",
      },
    }
  }
>
  <CheckBox
    checked={true}
    disabled={false}
    id="test"
    onChange={null}
  >
    <styled.div
      onClick={[Function]}
    >
      <div
        className="c0"
        onClick={[Function]}
      >
        <styled.div
          checked={true}
          disabled={false}
          hasFocus={false}
        >
          <div
            checked={true}
            className="c1"
            disabled={false}
          >
            <Styled(Icon)
              color="hsl(200, 74%, 46%)"
              height="1rem"
              name="check-mark"
              width="1rem"
            >
              <Icon
                className=""
                color="hsl(200, 74%, 46%)"
                height="1rem"
                name="check-mark"
                spin={false}
                width="1rem"
              >
                <styled.div
                  spin={false}
                >
                  <div
                    className="c2"
                  >
                    <svg
                      height="1rem"
                      viewBox="0 0 512 512"
                      width="1rem"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M411.582 134.248c8.33-8.33 21.838-8.33 30.17 0 8.33 8.332 8.33 21.84 0 30.17L207.085 399.085c-8.331 8.331-21.839 8.331-30.17 0L70.248 292.418c-8.33-8.33-8.33-21.838 0-30.17 8.332-8.33 21.84-8.33 30.17 0L192 353.83l219.582-219.582z"
                        fill="hsl(200, 74%, 46%)"
                        fillRule="evenodd"
                      />
                    </svg>
                  </div>
                </styled.div>
              </Icon>
            </Styled(Icon)>
            <input
              checked={true}
              disabled={false}
              id="test"
              onBlur={[Function]}
              onFocus={[Function]}
              tabIndex="0"
              type="checkbox"
            />
          </div>
        </styled.div>
        <styled.label
          disabled={false}
          hasFocus={false}
          htmlFor="test"
        >
          <label
            className="c3"
            disabled={false}
            htmlFor="test"
          />
        </styled.label>
      </div>
    </styled.div>
  </CheckBox>
</ThemeProvider>
`;

exports[`<CheckBox /> should render without a problem when unchecked and disabled 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 1.4rem;
  height: 1.4rem;
  border: 1px solid hsl(200,74%,46%);
  border-radius: 0.2rem;
  background: hsl(0,100%,100%);
  opacity: 0.4;
}

.c2 {
  display: block;
  margin-left: 0.7rem;
  color: hsl(213,17%,20%);
  opacity: 0.4;
}

<ThemeProvider
  theme={
    Object {
      "breakpoints": Object {
        "lg": 992,
        "md": 768,
        "sm": 576,
        "xl": 1200,
        "xs": 0,
      },
      "dimensions": Object {
        "big": "3.1rem",
        "bigInt": 3.1,
        "huge": "4.5rem",
        "hugeInt": 4.5,
        "medium": "1.5rem",
        "mediumInt": 1.5,
        "radius": "0.4rem",
        "radiusInt": 0.4,
        "small": "0.7rem",
        "smallInt": 0.7,
        "tiny": "0.3rem",
        "tinyInt": 0.3,
      },
      "fonts": Object {
        "family": "\\"Open Sans\\", Helvetica, Arial, sans-serif",
        "size": Object {
          "big": "3.2rem",
          "bigInt": 3.2,
          "huge": "6.4rem",
          "hugeInt": 6.4,
          "medium": "1.2rem",
          "mediumInt": 1.2,
          "small": "0.7rem",
          "smallInt": 0.7,
          "tiny": "0.3rem",
          "tinyInt": 0.3,
        },
        "spacing": Object {
          "big": "1.5px",
          "medium": "1px",
          "small": "0.5px",
        },
        "weight": Object {
          "normal": 400,
          "thick": 600,
          "thin": 200,
        },
      },
      "palette": Object {
        "alizarin": "hsl(6, 78%, 57%)",
        "anthracite": "hsl(213, 17%, 20%)",
        "black": "hsl(0, 0%, 0%)",
        "blackLight": "hsl(213, 17%, 20%)",
        "blue": "hsl(200, 74%, 46%)",
        "bodyBackground": "hsl(218, 29%, 95%)",
        "carrot": "hsl(28, 80%, 52%)",
        "clay": "hsl(215, 17%, 26%)",
        "darkBlue": "#1c88bf",
        "darkGray": "#dbdfe6",
        "darkGreen": "#7fbf3b",
        "darkRed": "#ea6151",
        "emerald": "hsl(145, 63%, 49%)",
        "fadeBlue": "hsl(210, 50%, 98%)",
        "failure": Object {
          "back": "hsl(6, 79%, 65%)",
          "darkBack": "#ea6151",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#f1978d",
          "lightFore": "#fff",
        },
        "googleBrandRed": "#ea4335",
        "gray": "hsl(218, 18%, 91%)",
        "green": "hsl(89, 53%, 52%)",
        "grey": "hsl(206, 16%, 60%)",
        "greygoose": "hsl(216, 29%, 97%)",
        "lightBlue": "#3baae2",
        "lightGray": "#fff",
        "lightGreen": "#a0d16b",
        "lightRed": "#f1978d",
        "marble": "hsl(206, 23%, 69%)",
        "orange": "hsl(34, 91%, 65%)",
        "primary": Object {
          "back": "hsl(200, 74%, 46%)",
          "darkBack": "#1c88bf",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#3baae2",
          "lightFore": "#fff",
        },
        "red": "hsl(6, 79%, 65%)",
        "river": "hsl(204, 70%, 53%)",
        "sand": "hsl(214, 8%, 84%)",
        "secondary": Object {
          "back": "hsl(0, 100%, 100%)",
          "darkBack": "#f7f7f7",
          "darkFore": "#8093a3",
          "fore": "hsl(206, 16%, 60%)",
          "lightBack": "#fff",
          "lightFore": "#a6b4be",
        },
        "success": Object {
          "back": "hsl(89, 53%, 52%)",
          "darkBack": "#7fbf3b",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#a0d16b",
          "lightFore": "#fff",
        },
        "turquoise": "hsl(168, 76%, 42%)",
        "white": "hsl(0, 100%, 100%)",
        "wisteria": "hsl(282, 44%, 47%)",
      },
    }
  }
>
  <CheckBox
    checked={false}
    disabled={true}
    id="test"
    onChange={null}
  >
    <styled.div
      onClick={[Function]}
    >
      <div
        className="c0"
        onClick={[Function]}
      >
        <styled.div
          checked={false}
          disabled={true}
          hasFocus={false}
        >
          <div
            checked={false}
            className="c1"
            disabled={true}
          >
            <input
              checked={false}
              disabled={true}
              id="test"
              onBlur={[Function]}
              onFocus={[Function]}
              tabIndex="0"
              type="checkbox"
            />
          </div>
        </styled.div>
        <styled.label
          disabled={true}
          hasFocus={false}
          htmlFor="test"
        >
          <label
            className="c2"
            disabled={true}
            htmlFor="test"
          />
        </styled.label>
      </div>
    </styled.div>
  </CheckBox>
</ThemeProvider>
`;

exports[`<CheckBox /> should render without a problem when unchecked and enabled 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 1.4rem;
  height: 1.4rem;
  border: 1px solid hsl(200,74%,46%);
  border-radius: 0.2rem;
  background: hsl(0,100%,100%);
}

.c2 {
  display: block;
  margin-left: 0.7rem;
  color: hsl(213,17%,20%);
}

<ThemeProvider
  theme={
    Object {
      "breakpoints": Object {
        "lg": 992,
        "md": 768,
        "sm": 576,
        "xl": 1200,
        "xs": 0,
      },
      "dimensions": Object {
        "big": "3.1rem",
        "bigInt": 3.1,
        "huge": "4.5rem",
        "hugeInt": 4.5,
        "medium": "1.5rem",
        "mediumInt": 1.5,
        "radius": "0.4rem",
        "radiusInt": 0.4,
        "small": "0.7rem",
        "smallInt": 0.7,
        "tiny": "0.3rem",
        "tinyInt": 0.3,
      },
      "fonts": Object {
        "family": "\\"Open Sans\\", Helvetica, Arial, sans-serif",
        "size": Object {
          "big": "3.2rem",
          "bigInt": 3.2,
          "huge": "6.4rem",
          "hugeInt": 6.4,
          "medium": "1.2rem",
          "mediumInt": 1.2,
          "small": "0.7rem",
          "smallInt": 0.7,
          "tiny": "0.3rem",
          "tinyInt": 0.3,
        },
        "spacing": Object {
          "big": "1.5px",
          "medium": "1px",
          "small": "0.5px",
        },
        "weight": Object {
          "normal": 400,
          "thick": 600,
          "thin": 200,
        },
      },
      "palette": Object {
        "alizarin": "hsl(6, 78%, 57%)",
        "anthracite": "hsl(213, 17%, 20%)",
        "black": "hsl(0, 0%, 0%)",
        "blackLight": "hsl(213, 17%, 20%)",
        "blue": "hsl(200, 74%, 46%)",
        "bodyBackground": "hsl(218, 29%, 95%)",
        "carrot": "hsl(28, 80%, 52%)",
        "clay": "hsl(215, 17%, 26%)",
        "darkBlue": "#1c88bf",
        "darkGray": "#dbdfe6",
        "darkGreen": "#7fbf3b",
        "darkRed": "#ea6151",
        "emerald": "hsl(145, 63%, 49%)",
        "fadeBlue": "hsl(210, 50%, 98%)",
        "failure": Object {
          "back": "hsl(6, 79%, 65%)",
          "darkBack": "#ea6151",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#f1978d",
          "lightFore": "#fff",
        },
        "googleBrandRed": "#ea4335",
        "gray": "hsl(218, 18%, 91%)",
        "green": "hsl(89, 53%, 52%)",
        "grey": "hsl(206, 16%, 60%)",
        "greygoose": "hsl(216, 29%, 97%)",
        "lightBlue": "#3baae2",
        "lightGray": "#fff",
        "lightGreen": "#a0d16b",
        "lightRed": "#f1978d",
        "marble": "hsl(206, 23%, 69%)",
        "orange": "hsl(34, 91%, 65%)",
        "primary": Object {
          "back": "hsl(200, 74%, 46%)",
          "darkBack": "#1c88bf",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#3baae2",
          "lightFore": "#fff",
        },
        "red": "hsl(6, 79%, 65%)",
        "river": "hsl(204, 70%, 53%)",
        "sand": "hsl(214, 8%, 84%)",
        "secondary": Object {
          "back": "hsl(0, 100%, 100%)",
          "darkBack": "#f7f7f7",
          "darkFore": "#8093a3",
          "fore": "hsl(206, 16%, 60%)",
          "lightBack": "#fff",
          "lightFore": "#a6b4be",
        },
        "success": Object {
          "back": "hsl(89, 53%, 52%)",
          "darkBack": "#7fbf3b",
          "darkFore": "#f7f7f7",
          "fore": "hsl(0, 100%, 100%)",
          "lightBack": "#a0d16b",
          "lightFore": "#fff",
        },
        "turquoise": "hsl(168, 76%, 42%)",
        "white": "hsl(0, 100%, 100%)",
        "wisteria": "hsl(282, 44%, 47%)",
      },
    }
  }
>
  <CheckBox
    checked={false}
    disabled={false}
    id="test"
    onChange={null}
  >
    <styled.div
      onClick={[Function]}
    >
      <div
        className="c0"
        onClick={[Function]}
      >
        <styled.div
          checked={false}
          disabled={false}
          hasFocus={false}
        >
          <div
            checked={false}
            className="c1"
            disabled={false}
          >
            <input
              checked={false}
              disabled={false}
              id="test"
              onBlur={[Function]}
              onFocus={[Function]}
              tabIndex="0"
              type="checkbox"
            />
          </div>
        </styled.div>
        <styled.label
          disabled={false}
          hasFocus={false}
          htmlFor="test"
        >
          <label
            className="c2"
            disabled={false}
            htmlFor="test"
          />
        </styled.label>
      </div>
    </styled.div>
  </CheckBox>
</ThemeProvider>
`;
